### PR TITLE
Add null checks for possible null trait

### PR DESCRIPTION
**Pull request by miguelpdiaz8**
Adding null checks since it is possible the trait is null and we are calling .trim() function on it

## Testing

Added unit tests for all use cases and I'll test it on staging manually with Ria

### CODE REVIEW

File: packages/destination-actions/src/destinations/engage/sendgrid/__tests__/send-email.test.ts [hunk 1]
```diff
@@ -1123,15 +1151,96 @@ describe.each([
...
+              id: userData.email,
+              isSubscribed: true,
+              type: 'email'
+            }
+          ]
+        }),
+        settings,
+        mapping: getDefaultMapping({
+          subject: 'Hi {{profile.traits.lastName | default: "Person"}}',
+          traits: {
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage/sendgrid/__tests__/send-email.test.ts:1229]: `traits` as an object can be null for both email and sms. lets add protection to that

**Comment by alfrimpong** [comment]
Lets bring in additional reviewers from both our team and the CP team to take a look at this too.

File: packages/destination-actions/src/destinations/engage/sendgrid/__tests__/send-email.test.ts (hunk 1 above)
Comment (miguelpdiaz8) [reply packages/destination-actions/src/destinations/engage/sendgrid/__tests__/send-email.test.ts:1229]: done

File: packages/destination-actions/src/destinations/engage/twilio/utils/TwilioMessageSender.ts [hunk 2]
```diff
@@ -38,16 +38,25 @@ export abstract class TwilioMessageSender<TPayload extends TwilioPayloadBase> ex
     content: R,
     profile: Profile
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage/twilio/utils/TwilioMessageSender.ts:39]: Lets change the type of `Profile.traits` from `Record<string, string>` to `Record<string, any>`

**Review by alfrimpong — CHANGES_REQUESTED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @miguelpdiaz8 and @alfrimpong  - do you still want to get this code deployed?

**Comment by miguelpdiaz8** [comment]
> hi @miguelpdiaz8 and @alfrimpong - do you still want to get this code deployed?

Hi @joe-ayoub-segment I already tested this, and it would be great to deploy it tomorrow but Albert is OOO today and I need his review since he requested a change, at what time are you deploying tomorrow?

**Comment by joe-ayoub-segment** [comment]
Hi @miguelpdiaz8 I'll be starting the deploy in about 1h from now.

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts [hunk 3]
```diff
@@ -90,10 +90,19 @@ export class SendEmailPerformer extends MessageSendPerformer<Settings, Payload>
     },
     contentType: string
   ) {
+    const traits = liquidData.profile.traits ? { ...liquidData.profile.traits } : liquidData.profile.traits
```
Comment (alfrimpong) [inline packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:93]: Nonblocking Nit: This conditional feels a bit weird. `traits = {...liquidData.profile.traits}` instead?

**Review by alfrimpong — APPROVED** [review]

**Comment by alfrimpong** [comment]
> Hi @miguelpdiaz8 I'll be starting the deploy in about 1h from now. Can you and @alfrimpong please confirm that this code is good to go?

Its good

File: packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts (hunk 3 above)
Comment (alfrimpong) [reply packages/destination-actions/src/destinations/engage/sendgrid/sendEmail/SendEmailPerformer.ts:93]: Eh, its actually fine. protects in the case of `undefined` (though it shouldnt be possible, we really never know)

**Comment by joe-ayoub-segment** [comment]
@alfrimpong indicated on Slack that this PR is good to go.

**Comment by joe-ayoub-segment** [comment]
hi @miguelpdiaz8 and @alfrimpong PR deployed!

**Pull request merged**